python/dgl/transform.py

```diff
@@ -673,71 +676,693 @@ def metapath_reachable_graph(g, metapath):
+        The type of the edges to remove. Can be omitted if there is
+        only one edge type in the graph.
+
+    Return
+    ------
+    DGLHeteroGraph
+        The graph with edges deleted.
 
     Examples
-    ---------
+    --------
+    >>> import dgl
+    >>> import torch
+
+    **Homogeneous Graphs or Heterogeneous Graphs with A Single Edge Type**
+
+    >>> g = dgl.graph((torch.tensor([0, 0, 2]), torch.tensor([0, 1, 2])))
+    >>> g.edata['he'] = torch.arange(3).float().reshape(-1, 1)
+    >>> g = dgl.remove_edges(g, torch.tensor([0, 1]))
+    >>> g
+    Graph(num_nodes=3, num_edges=1,
+        ndata_schemes={}
+        edata_schemes={'he': Scheme(shape=(1,), dtype=torch.float32)})
+    >>> g.edges('all')
+    (tensor([2]), tensor([2]), tensor([0]))
+    >>> g.edata['he']
+    tensor([[2.]])
+
+    **Heterogeneous Graphs with Multiple Edge Types**
+
+    >>> g = dgl.heterograph({
+    >>>     ('user', 'plays', 'game'): (torch.tensor([0, 1, 1, 2]),
+    >>>                                 torch.tensor([0, 0, 1, 1])),
+    >>>     ('developer', 'develops', 'game'): (torch.tensor([0, 1]),
+    >>>                                         torch.tensor([0, 1]))
+    >>>     })
+    >>> g = dgl.remove_edges(g, torch.tensor([0, 1]))
+    DGLError: Edge type name must be specified
+    if there are more than one edge types.
+    >>> g = dgl.remove_edges(g, torch.tensor([0, 1]), 'plays')
+    >>> g.edges('all', etype='plays')
+    (tensor([0, 1]), tensor([0, 0]), tensor([0, 1]))
+    See Also
+    --------
+    add_nodes
+    add_edges
+    remove_nodes
+    """
+    # TODO(xiangsx): block do not support remove_edges
+    if etype is None:
+        if len(g.etypes) != 1:
+            raise DGLError('Edge type name must be specified if there are more than one ' \
+                            'edge types.')
+    eids = utils.prepare_tensor(g, eids, 'u')
+    assert g.number_of_edges(etype) > F.as_scalar(F.max(eids, dim=0)), \
+        'The input eid {} is out of the range [0:{})'.format(
+            F.as_scalar(F.max(eids, dim=0)), g.number_of_edges(etype))
+
+    # edge_subgraph
+    edges = {}
+    etype = g.to_canonical_etype(etype)
+    for c_etype in g.canonical_etypes:
+        # the target edge type
+        if c_etype == etype:
+            old_eids = g.edges(form='eid', order='eid', etype=c_etype)
+            # trick here, eid_0 is 0 and should be handled
+            old_eids = F.scatter_row(old_eids,
+                                     F.copy_to(F.tensor(0, dtype=F.int64),
+                                               F.context(old_eids)),
+                                     F.copy_to(F.tensor(1, dtype=F.dtype(old_eids)),
+                                               F.context(old_eids)))
+            old_eids = F.scatter_row(old_eids, eids, F.full_1d(
+                len(eids), 0, F.dtype(old_eids), F.context(old_eids)))
+            edges[c_etype] = F.tensor(F.nonzero_1d(old_eids), dtype=F.dtype(old_eids))
```

Comment:
Suggest having a `utils.compensate(ids, max_range)` for computing the compensate set of the given IDs.